### PR TITLE
Py3 support

### PR DESCRIPTION
**Pull request by jburel**
* Upgrade notebook
* Upgrade dependencies
Dependencies file and Docker to be reviewed when https://github.com/IDR/jupyter-docker/pull/36 is merged and image available

to test: 
* replace ``conn = connection()`` by ``conn = connection('test80', 'public', 'public')``

### CODE REVIEW

File: docker/environment-python3-idr.yml [hunk 1]
```diff
@@ -15,9 +12,9 @@ dependencies:
 - python-graphviz=0.13=py_0
 - python-igraph=0.7.1.post6=py36_4
 - scikit-learn=0.21.*
-- seaborn=0.9.0=py36_0
 - tqdm=4.36.1=py_0
+- pip
 # defaults
 - pip:
   - py2cytoscape==0.7.1
```
Comment (manics) [inline docker/environment-python3-idr.yml]: Looks like this is now on conda-forge: https://anaconda.org/conda-forge/py2cytoscape

Comment (jburel) [reply docker/environment-python3-idr.yml]: thanks

File: docker/environment-python3-idr.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+name: python3
+channels:
+- conda-forge
```
Comment (manics) [inline docker/environment-python3-idr.yml]:
```suggestion
- ome
- manics
- conda-forge
```

File: docker/environment-python3-idr.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+- bokeh=1.4.0=py36_0
+- joblib=0.12.1=py36_0
+- markdown=3.1=py36_0
+- python-graphviz=0.13=py_0
+- python-igraph=0.7.1.post6=py36_4
+- scikit-learn=0.21.*
+- tqdm=4.36.1=py_0
+- py2cytoscape==0.7.1
+# defaults
+- pip:
```
Comment (manics) [inline docker/environment-python3-idr.yml]:
```suggestion
```

File: docker/environment-python3-idr.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  - idr-py>=0.4.0.dev3
```
Comment (manics) [inline docker/environment-python3-idr.yml]:
```suggestion
```

File: docker/environment-python3-idr.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+name: python3
+channels:
+- conda-forge
+- defaults
+- bioconda
+dependencies:
+- gseapy=0.9.8=py36_0
+- pip
```
Comment (manics) [inline docker/environment-python3-idr.yml]:
```suggestion
```

File: docker/environment-python3-idr.yml [hunk 6]
```diff
@@ -0,0 +1,20 @@
+name: python3
+channels:
+- conda-forge
+- defaults
+- bioconda
+dependencies:
+- gseapy=0.9.8=py36_0
```
Comment (manics) [inline docker/environment-python3-idr.yml]:
```suggestion
# ome (todo)
- idr-py
# bioconda
- gseapy=0.9.8=py36_0
```

**Comment by dominikl** [comment]
Tested all notebooks.

CalculateSharpness.ipynb: Pass
Note: Should be mentioned that it needs idr0009-simpson-secretion/screenA .
With plates from other datasets you get various index out of bounds erros,
as the notebook seems to a assume a certain plate layout / number of channels.

Figure_1_Sampling_of_Phenotypes.ipynb: Pass

GeneNetwork.ipynb: Errors
Changes needed in last code block:
names = list(pts.keys())
cntr = 0
for x in pts.values():

GenesToPhenotypes.ipynb: Pass

Getting_Started.ipynb: Pass

IDR_API_example_script.ipynb: Errors
Section "Get Image Thumbnail:"
Fix: for iid, thumb in json.items():

PCAanalysisOfCharmFeatures.ipynb: ?
Takes too long to run

QueryIDRWithGeneLists.ipynb: Pass

RohnPhenotypeClustering.ipynb: Errors
Various errors (key error, value error, etc.),
although imho not python3 related.

SysgroOverview.ipynb: Errors
Errors like:
AttributeError: 'FileAnnotationI' object has no attribute 'getMapValueAsMap'
Probably not python3 relaetd.

SysgroRoiLength.ipynb: Pass

**Note: The two python3 related issues are fixed in: https://github.com/jburel/idr-notebooks/pull/3**

**Comment by dominikl** [comment]
Thanks. Both notebooks work now. RohnPhenotypeClustering and SysgroOverview still have errors, but I think they are not python3 related. 👍

**Comment by jburel** [comment]
Thanks
I will look at conda tomorrow. if we are running out of time we can always install from pypi.

**Comment by dominikl** [comment]
Tested again, works 👍

**Pull request merged**